vendor/bam/src/json/frame.rs

```diff
@@ -272,22 +192,23 @@ impl JsonFrameHandler {
         warn!("Failed to dispatch request to handler because: {:?}", error);
 
         match error {
-            RequestError::UnknownMandatoryHeaders(header_keys) => {
-                response.with_header("Unsupported-Headers", header_keys)
-            }
+            RequestError::UnknownMandatoryHeaders(header_keys) => response.with_header(
+                "Unsupported-Headers",
+                Header::with_value(header_keys)
+                    .expect("list of strings should serialize to serde_json::Value"),
+            ),
             _ => response,
         }
     }
 }
 
 #[derive(Debug, Clone, PartialEq)]
-enum HeaderErrorKind {
+pub(crate) enum HeaderErrorKind {
     UnknownMandatoryHeader,
-    DecodingError,
 }
 
 #[derive(Debug, Clone)]
-struct HeaderError {
+pub(crate) struct HeaderError {
```

Comment:
⁉️are all the `pub(crate)` necessary? It doesn't seem like something we couldn't just make `pub`.